sources/imdsclient/src/lib.rs

```diff
@@ -627,14 +638,17 @@ mod test {
     async fn fetch_token_timeout() {
         let server = Server::run();
         let base_uri = format!("http://{}", server.addr());
+        let retry_token_timeout = Duration::from_secs(10);
```

Comment:
Am I correct in understanding that this test expects to wait this full timeout value, confirming that it does fail after that time?  If so, this seems like quite a long time to block a unit test process.  Can we do 1s?

Reply:
I dropped it to 2s since the other tests take about 2s on my machine.